src/relations/db.py

```diff
@@ -414,10 +462,14 @@ def _on_relation_broken(self, broken_event: RelationBrokenEvent):
         if not self._check_backend() or None in [user, database]:
             # this relation was never created, so wait for it to be initialised before removing
             # everything.
-            logger.warning(
-                f"backend relation not yet available - deferring {self.relation_name}-relation-broken event."
-            )
-            broken_event.defer()
```

Comment:
This caused a perpetual deferral with extensions, as the joining never completed.